elastic/fleet-server
# replace secret refs

## What is the problem this PR solves?

Agent policy secrets are sent to agents in agent policy.

## How does this PR solve the problem?

On checkin, read secret values of secret references in the agent policy, and replace occurrences in input and stream variables.

## How to test this PR locally

- Start local es, kibana, fleet server
- Enable secrets feature flag in `kibana.dev.yml`: `xpack.fleet.enableExperimental: ['secretsStorage']`
- Start local registry with secret package here: https://github.com/elastic/kibana/pull/157176
- Open kibana, add integration: secret package, add values for package, input and stream secret vars
- When the agent policy with the secret integration is created, the agent policy should have 3 secret references
- Enroll an agent locally
- Download diagnostics and check in `computed-config.yml` that the inputs contain the secret values, not references
- Note: the agent is unhealthy with the error message `input not supported` is not related to secrets, looks like a validation [changed](https://github.com/elastic/elastic-agent/commit/64fc9d91dbea21ab51a9a72dcc12e4b42b723854#diff-db2140f630be8cde1ea80e76eff1c58c5a9d95022bff853724e205ee9fe58d9f) on agent side, so that the test package doesn't have a valid input type. Can be ignored for now.

Example:
```
inputs:
- data_stream:
    namespace: default
  id: test_input-secrets-7c6b08d8-dbe5-4278-8929-dfb26ea8e579
  input_var_secret: input
  meta:
    package:
      name: secrets
      version: 1.0.0
  name: secrets-1
  package_policy_id: 7c6b08d8-dbe5-4278-8929-dfb26ea8e579
  package_var_secret: pkg
  policy:
    revision: 5
  revision: 1
  streams:
  - config.version: "2"
    data_stream:
      dataset: secrets.log
      type: logs
    id: test_input-secrets.log-7c6b08d8-dbe5-4278-8929-dfb26ea8e579
    input_var_secret: input
    package_var_secret: pkg
    stream_var_secret: stream
  type: test_input
```

## Design Checklist




## Checklist





## Related issues

- Closes https://github.com/elastic/fleet-server/issues/2485
